frontend/src/metabase-lib/parameters/utils/targets.ts

```diff
@@ -1,14 +1,38 @@
-import type { ParameterTarget } from "metabase-types/api";
+import type {
+  ExpressionReference,
+  FieldReference,
+  LocalFieldReference,
+  NativeParameterDimensionTarget,
+  ParameterTarget,
+  ParameterTextTarget,
+  ParameterVariableTarget,
+  StructuredParameterDimensionTarget,
+} from "metabase-types/api";
 import { isDimensionTarget } from "metabase-types/guards";
 import * as Lib from "metabase-lib";
+import type { TemplateTagDimension } from "metabase-lib/Dimension";
 import Dimension from "metabase-lib/Dimension";
 import type Question from "metabase-lib/Question";
 import type StructuredQuery from "metabase-lib/queries/StructuredQuery";
 import type NativeQuery from "metabase-lib/queries/NativeQuery";
 import type TemplateTagVariable from "metabase-lib/variables/TemplateTagVariable";
 
-export function isVariableTarget(target: ParameterTarget) {
-  return target?.[0] === "variable";
+export function isParameterVariableTarget(
+  target: ParameterTarget,
+): target is ParameterVariableTarget {
+  return target[0] === "variable";
+}
+
+function isConcreteFieldReference(
+  reference: FieldReference,
+): reference is LocalFieldReference {
+  return reference[0] === "field";
+}
```

Comment:
`LocalFieldReference` must have a numeric ID. There are field references that use a column name when the ID isn't deterministic (for example when referencing columns from a native query, one of the previous stages, etc.):

```js
[ "field", "CREATED_AT', { "base-type": "type/DateTime" } ]
```

If we want _only_ specific field references, we should update the type-guard to check the ID type
If we're okay with both kinds of references, we should do something else here